components/opportunity-detail.tsx

```diff
@@ -1,94 +1,61 @@
 "use client";
 import { formatDistanceToNow } from 'date-fns';
 import { CircleUserRound } from "lucide-react";
-import { CredentialStatus, DecodedJWT, DecodedRequestedCredential, Opportunity } from '@/lib/types';
+import { Opportunity } from '@/lib/types';
 import CredentialBadge from '@/components/credential-badge';
 import ReactMarkdown from 'react-markdown';
 import { Card, CardContent, CardFooter } from '@/components/ui/card';
 import { Button } from '@/components/ui/button';
 import requestVerifiableCredential from '@/lib/services/requestCredential';
 import { useAuth } from '../lib/context/AuthContext';
-import { jwtDecode } from 'jwt-decode';
-import { useState } from 'react';
-import RequestCredentialModal from '@/components/ui/RequestCredentialModal'; // Import the new ModalOverlay component
+import { useEffect, useState } from 'react';
 
 interface OpportunityDetailProps {
   opportunity: Opportunity;
 }
 
 export default function OpportunityDetail({ opportunity }: OpportunityDetailProps) {
-  const [modalOpen, setModalOpen] = useState(false);
-  const [statusMessage, setStatusMessage] = useState('');
-  const [credentialStatus, setCredentialStatus] = useState<CredentialStatus>(null);
-  
   const formattedDate = formatDistanceToNow(new Date(opportunity.postedDate), { addSuffix: true });
   const { principal } = useAuth();
+  const [userPermission, setUserPermission] = useState<boolean>(false);
 
-  function getCredentialFromLocalStorage(key: string) {
-    try {
-      const storedCredential: string | null = localStorage.getItem(key);
-      if (!storedCredential) return null;
-
-      const decodedToken: DecodedJWT = jwtDecode(storedCredential);
-      const decodedIssuerToken: DecodedRequestedCredential = jwtDecode(decodedToken.vp.verifiableCredential[1]);
-      const expTimeStamp = decodedIssuerToken.exp;
-
-      if (new Date() > new Date(expTimeStamp * 1000)) {
-        localStorage.removeItem(key);
-        return null;
-      }
-      return storedCredential;
-    } catch (error) {
-      console.log(error);
-      return null;
-    }
-  }
-
-  async function handleCheckPermissions() {
+  useEffect(() => {
     if (!principal) {
-      alert("Login to check permissions");
+      setUserPermission(false);
       return;
-    }
+    } 
 
     const credentialId = opportunity.requiredCredentials[0].id;
     const key = `credential_${principal}_${credentialId}`;
-
-    const credential = getCredentialFromLocalStorage(key);
-    if (credential) {
-      setCredentialStatus("present");
-      setStatusMessage("Credential present");
-      setModalOpen(true);
-      return;
+    if (getCredentialStatus(key)) {
+      setUserPermission(true);
+    } else {
+      setUserPermission(false);
+    } 
+  }, [principal]);
+
+  function getCredentialStatus(key: string): boolean {
+    try {
+      return localStorage.getItem(key) ? true : false;
+    } catch {
+      return false;
     }
```

Comment:
This approach use access the localstorage only if the window exists, and instead of the ternany if and else,  we can just make the result a boolean value, if the window is undefined then the returned value will be false, if the key is null it's going to be false as well.

```suggestion
  function getCredentialStatus(key: string): boolean {
      return !!window?.localStorage.getItem(key) 
```